server/tests/users.js

```diff
@@ -0,0 +1,168 @@
+  it('Should return status code 200 and message', (done) => {
+    chai.request(app)
+      .get('/')
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('message');
+        expect(res.body.message).to.equal('Welcome to AnnouneIT!');
+        done();
+      });
+  });
+});
+
+describe('User sign up - valid', () => {
+  it('Should return status code 201', (done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signup')
+      .send({
+        firstname: 'John',
+        lastname: 'Mugabo',
+        email: 'mugabo@gmail.com',
+        phone: '+250787770000',
+        address: 'Kigali, Rwanda',
+        password: 'mugabojohn',
+        confirmpassword: 'mugabojohn',
+      })
+      .end((err, res) => {
+        expect(res.body).to.have.property('status');
+        expect(res.status).to.equal(201);
+        expect(res.body).to.have.property('message');
+        expect(res.body.message).to.equal('User created successfully');
+        expect(res.body).to.have.property('data');
+        expect(res.body.data).to.have.property('id');
+        expect(res.body.data).to.have.property('firstname');
+        expect(res.body.data).to.have.property('lastname');
+        expect(res.body.data).to.have.property('email');
+        expect(res.body.data).to.have.property('phone');
+        expect(res.body.data).to.have.property('address');
+        expect(res.body.data).to.have.property('isAdmin');
+        expect(res.body.data).to.have.property('status');
+        expect(res.body.data).to.have.property('registered');
+        done();
+      });
+  });
+});
+
+describe('User sign up - invalid firstname', () => {
+  it('Should return status code 400', (done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signup')
+      .send({
+        lastname: 'Mugabo',
+        email: 'mugabo@gmail.com',
+        phone: '+250787770000',
+        address: 'Kigali, Rwanda',
+        password: 'mugabojohn',
+        confirmpassword: 'mugabojohn',
+      })
+      .end((err, res) => {
+        expect(res.body).to.have.property('status');
+        expect(res.status).to.equal(400);
+        expect(res.body).to.have.property('error');
+        done();
+      });
+  });
+});
+
+describe('User sign up - passwords no match', () => {
+  it('Should return status code 400', (done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signup')
+      .send({
+        firstname: 'John',
+        lastname: 'Mugabo',
+        email: 'mugabo@gmail.com',
+        phone: '+250787770000',
+        address: 'Kigali, Rwanda',
+        password: 'mugabojohn',
+        confirmpassword: 'hellothere',
+      })
+      .end((err, res) => {
+        expect(res.body).to.have.property('status');
+        expect(res.status).to.equal(400);
+        expect(res.body).to.have.property('error');
+        expect(res.body.error).to.equal('Please make sure your passwords are matching');
```

Comment:
If I do remember we agreed to put all messages in a separate file and use it wherever needed

Reply:
Yes we did, I'm working on it now. Thanks for reminding me.